tensordict/prototype/tensorclass.py

```diff
@@ -123,14 +148,34 @@ def __torch_function__(
         args: tuple[Any, ...] = (),
         kwargs: dict[str, Any] | None = None,
     ) -> Callable:
+        if (
```

Comment:
What happens to the metadata in all these cases? Is it copied from the original tensordict?
If this is too brittle we can only take care of the `tensorclass.method` and not the overridable torch functions.
We should also consider if this implementation does not cause a (small) overhead when scanning the input and output types in search of the strategy, and whether it won't break on some torch function we have not implemented yet.

I'm trying to understand what other torch function we may want to implement in the future.
To get a complete list of overridable functions run `torch.overrides.get_overridable_functions()`
There are some that we may want to implement like `index_fill`, most of these are algebraic ops though.

Reply:
Ok then if you feel like this is something we can work with I'm happy to merge it

Reply:
I think this amount of overhead is acceptable. The alternative would mean reintroducing a lot of the overrides that I've just deleted. There's a bit of a trade-off between having small amount of generic code with some overhead vs. a large amount of specific code.

The only question then is whether you're happy with how we handle the metadata. I'm not sure what the alternatives are tbh, and afaict this PR is at least not changing behaviour.